lib/ain-evm/src/executor.rs

```diff
@@ -246,6 +249,67 @@ impl<'backend> AinExecutor<'backend> {
         ))
     }
 
+    pub fn exec_with_tracer(
+        &mut self,
+        signed_tx: &SignedTx,
+    ) -> Result<(Vec<ExecutionStep>, bool, Vec<u8>, u64)> {
+        let to = signed_tx.to().ok_or(format_err!(
+            "debug_traceTransaction does not support contract creation transactions",
+        ))?;
+        self.backend.update_vicinity_from_tx(signed_tx)?;
+        trace!(
+            "[Executor] Executing trace EVM TX with vicinity : {:?}",
+            self.backend.vicinity
+        );
+        let ctx = ExecutorContext {
+            caller: signed_tx.sender,
+            to: Some(to),
+            value: signed_tx.value(),
+            data: signed_tx.data(),
+            gas_limit: u64::try_from(signed_tx.gas_limit())?, // Sets to u64 if overflows
```

Comment:
Don't we want to unwrap_or(u64::MAX) in line with the comment ?

Reply:
The comments were inaccurate, my bad. We should propagate the overflow error (which shouldnt ever happen) back to the RPC call.